src/Compilers/CSharp/Test/Semantic/Semantics/NullableReferenceTypesTests.cs

```diff
@@ -52748,6 +52749,215 @@ static void F3(object? x3, object y3)
                 Diagnostic(ErrorCode.WRN_NullReferenceReceiver, "z2").WithLocation(22, 9));
         }
 
+        [Fact]
+        [WorkItem(48174, "https://github.com/dotnet/roslyn/issues/48174")]
+        public void Lambda_Nesting_Large_01()
+        {
+            const int depth = 30;
+            var builder = new StringBuilder();
+            for (int i = 1; i < depth; i++)
+            {
+                builder.AppendLine($"        M0(c{i} =>");
+            }
+            builder.Append($"        M0(c{depth} => {{ }}");
+            for (int i = 0; i < depth; i++)
+            {
+                builder.Append(")");
+            }
+            builder.Append(";");
+            var source = @"
+using System;
+
+class C
+{
+    void M0(Action<C> action) { }
+
+    void M1()
+    {
+" + builder.ToString() + @"
+    }
+}
+";
+            var comp = CreateNullableCompilation(source);
+            comp.VerifyDiagnostics();
+        }
+
+        [Fact]
+        [WorkItem(48174, "https://github.com/dotnet/roslyn/issues/48174")]
+        public void Lambda_Nesting_Large_02()
+        {
+            const int depth = 30;
+            var builder = new StringBuilder();
+            for (int i = 0; i < depth; i++)
+            {
+                builder.AppendLine($"        Action<C> a{i} = c{i} => {{");
+            }
+            for (int i = 0; i < depth; i++)
+            {
+                builder.AppendLine("        };");
+            }
+            var source = @"
+using System;
+
+class C
+{
+    void M1()
+    {
+" + builder.ToString()
++ @"    }
+}
+";
+            var comp = CreateCompilation(source, options: WithNonNullTypesFalse());
```

Comment:
>WithNonNullTypesFalse() [](start = 58, length = 23)

`WithNonNullTypesTrue()`?

Reply:
Oops 😅 at least the test runtime is still fast when using `WithNonNullTypesTrue()` locally.